# Overview
In this notebook you will calculate a variety of descriptive statistics for a dataset comprised of four-year-old apple tree girth measurements in mm. Complete each cell in the notebook then save the results as an html file to submit for grading. 

* Run the following cell (shift-enter) to load needed python packages and modules.

In [4]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats

* Load appletrees.csv into a pandas dataframe.
* If this was in your library you would use the path `.../library/filename.csv`.
* Use the `.head()` method to print out the first 7 rows of the dataframe.
* Get the `.shape` (no parentheses) property to see how many rows and columns are in the dataset.

In [5]:
# DON'T MODIFY THIS CELL
url = "https://raw.githubusercontent.com/prof-groff/evns462/master/data/appletrees.csv"
appletrees = pd.read_csv(url)
print(appletrees.head(7))
print("shape: ", appletrees.shape)

   girth mm
0       108
1        99
2       106
3       102
4       115
5       120
6       120
shape:  (60, 1)


## Questions:

How many different apple trees (rows) were sampled for this dataset?

What is the measurement scale (or data type) of the girth data?

## Measures of Central Tendency 
* Calculate and print the mean, mode, and median of the apple tree girth sample data.
* Make a histogram of this data with axes labels.
* Do a Shapiro-Wilk test for normality on the data and print the test statistica and p-value.

In [11]:
# Hint: You can extract a specific column using dataframe["column_name"].
# Hint: The Shapiro-Wilk test for normality is included in the stats module as stats.shapiro(x)

## Questions:
How does the mean, mode, and median of the the data compare? 

What does this tell you about the distribution?

Does the histogram of the data look normally distriubted? Why or why not?

Does the Shapiro-Wilk test support or reject the null hypothesis that the data is normally distributed? This is tricky because a p value less than 0.05 means the null is rejected and the data is NOT normally distributed so you are "hopeing" for a large p value.


## Measures of Variability
* Calculate the sample variance and sample standard deviation of the apple tree girth data.
* Calculate the standard error of the mean apple tree girth.

In [13]:
# Hint: The degrees of freedom used in calculating a sample mean is N-1 so use np.std(x,ddof=1)

## Questions:
What is the meaning of the standard error of the mean?

If the number of trees in the sample were 120 instead of 60, how would you expect the standard error of the mean to change?